### PR TITLE
User/bsmithvt/idpss unavailable bug

### PR DESCRIPTION
**Pull request by bsmithvt**

### CODE REVIEW

File: Fabric.Identity.API/Infrastructure/PolicyProvider.cs [hunk 1]
```diff
@@ -11,7 +13,17 @@ public class PolicyProvider
         public readonly CircuitBreakerPolicy LdapErrorPolicy = Policy.Handle<LdapException>()
             .CircuitBreaker(5, TimeSpan.FromMinutes(5));
 
+        public readonly RetryPolicy IdPSearchServiceRetryPolicy = Policy.Handle<HttpRequestException>()
+            .RetryAsync(3);
+
         public readonly CircuitBreakerPolicy IdPSearchServiceErrorPolicy = Policy.Handle<HttpRequestException>()
-            .CircuitBreaker(5, TimeSpan.FromMinutes(5));
+            .CircuitBreakerAsync(3, TimeSpan.FromMinutes(1));
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Infrastructure/PolicyProvider.cs]: Is there a reason we want to close the circuit after one minute instead of 5?

Comment (bsmithvt) [reply Fabric.Identity.API/Infrastructure/PolicyProvider.cs]: I figured 5 minutes was a rather long time to leave the circuit open, especially based on how this bug surfaces, but I can switch it back.

**Comment by bsmithvt** [comment]
@michaelvidal24 I'm seeing almost immediate failure when testing with IdP SS unavailable, but I can set an explicit timeout.

**Comment by michaelvidal24** [comment]
@bsmithvt - I think the service unavailable is one scenario. But another is if the service is stuck trying to stat up and can't respond. The http request will hang waiting for a response, so I think we should try to handle that with a smallish timeout on the http request. Ideally, that timeout would only apply for these specific requests, and not set a shared http client timeout.

**Comment by bsmithvt** [comment]
@michaelvidal24 ok, good call. I wasn't aware that behavior was different (stopped vs. starting). I've updated the PR to specify an explicit timeout and handle exceptions associated with timeouts.

**Pull request merged**